libraries/helpers.rb

```diff
@@ -38,6 +38,59 @@ def habitat_plan_dir
       end
     end
 
+    #
+    # Search the workspace repo for any plan.sh files inside of habitat
+    # directories, and return an array of paths to the PLAN_CONTEXT for each
+    # service.
+    #
+    # Example:
+    # Let's assume you have multiple habitat PLAN_CONTEXTs in a single repository
+    #
+    # $ tree -d .
+    #   .
+    #   ├── service-one
+    #   |   ├── habitat
+    #   |   |   ├── plan.sh
+    #   ├── service-two
+    #   |   ├── habitat
+    #   |   |   ├── plan.sh
+    #   ├── habitat-plans
+    #   |   ├── nested-service
+    #   |   |   ├── plan.sh
+    #
+    # This helper will return the following array:
+    # => [
+    #     "/src/service-one",
+    #     "/src/service-two",
+    #     "/src/habitat-plans/nested-service"
+    #    ]
+    #
+    def all_habitat_plan_contexts
+      all_dirs = []
+
+      # Handle the legacy .delivery/config.json configuration
+      all_dirs << habitat_plan_dir
+
+      # Search the repository for any habitat PLAN_CONTEXTs
+      Dir.glob(File.join(workflow_workspace_repo, '**/**/plan.sh')).map do |plan_file|
+        path_components = Pathname(plan_file).dirname.each_filename.to_a
+        idx = path_components.last == 'habitat' ? -2 : -1
```

Comment:
This might be a little easier to read if this line were `path_components.pop if path_components.last == 'habitat'` and then the next line didn't do any slicing.